src/server/server_family.cc

```diff
@@ -426,7 +426,8 @@ void ServerFamily::Init(util::AcceptServer* acceptor, std::vector<facade::Listen
   string flag_dir = GetFlag(FLAGS_dir);
   if (IsCloudPath(flag_dir)) {
     aws_ = make_unique<cloud::AWS>("s3");
-    if (auto ec = aws_->Init(); ec) {
+    auto ec = shard_set->pool()->GetNextProactor()->Await([&] { return aws_->Init(); });
```

Comment:
also, next time you change something in helio, please also add a comment for `AWS::Init()` saying that the function should run in the proactor thread.